lib/media-controls-impl.js

```diff
@@ -0,0 +1,524 @@
+          window.removeEventListener('mouseup', this, true);
+          this.mouseEventHandlerRegistered = false;
+        }
+        break;
+
+    case 'loadedmetadata':
+      //
+      // Metadata has been loaded, now we can set the duration of the media
+      //
+      this.els.durationText.textContent = this.formatTime(this.mediaPlayer.duration);
+      break;
+
+    case 'play':
+      //
+      // Media is playing, display 'paused' icon
+      //
+      this.els.play.classList.remove('paused');
+
+      // Update l10n-id for the benefit of the screen reader
+      this.els.play.setAttribute('data-l10n-id', 'playbackPlay');
+      break;
+
+    case 'pause':
+      //
+      // Media is paused, display 'play' icon
+      //
+      this.els.play.classList.add('paused');
+
+      // Update l10n-id for the benefit of the screen reader
+      this.els.play.setAttribute('data-l10n-id', 'playbackPause');
+
+      // If the paused event comes when the media is at the end,
+      // set our 'played-till-end' flag. The one exception is when
+      // the paused event comes from pausing the media when the
+      // forward button was used to seek to the end while the media
+      // was playing. In this case, we don't consider the media being
+      // played until the end.
+      if (this.mediaPlayer.currentTime === this.mediaPlayer.duration) {
+        if (this.pausedAtEndWhilePlaying) {
+          this.pausedAtEndWhilePlaying = false;
+        }
+        else {
+          this.playedUntilEnd = true;
+        }
+      }
+      break;
+
+    case 'timeupdate':
+      //
+      // Update the progress bar and play head as the video plays
+      //
+      if (!this.mediaControlsElement.hidden) {
+        this.updateMediaControlSlider();
+      }
+      break;
+
+    case 'seeked':
+      //
+      // Update the position of the slider when the video position has been
+      // moved.
+      //
+      this.updateMediaControlSlider();
+      break;
+
+    case 'ended':
+      //
+      // Ignore ended events that occur while the user is dragging the slider
+      //
+      if (this.dragging) {
+        return;
+      }
+
+      // If the media was played until the end (as opposed to being forwarded
+      // to the end), position the player at the beginning of the video.
+      if (this.playedUntilEnd) {
+        this.mediaPlayer.currentTime = 0;
+        this.playedUntilEnd = false;
+      }
+      break;
+  }
+
+  if (e.target === this.els.sliderWrapper && (e.type === 'touchstart' || e.type === 'mousedown' ||
+           e.type === 'touchmove') ||
+           e.type === 'mousemove') {
+
+    var clientX =
+      (/mouse/.test(e.type)) ? e.clientX : e.changedTouches[0].clientX;
+
+    switch(e.type) {
+      case 'touchstart':
+      case 'mousedown':
+        if (e.type === 'mousedown') {
+          window.addEventListener('mousemove', this, true);
+          window.addEventListener('mouseup', this, true);
```

Comment:
You already registered these handlers up above in the first switch statement.  And you didn't set the flag here to remember that they are registered.  You may be registering more and more event handlers on each mouse down event...